jsapp/js/components/reports/questionGraphSettings.component.tsx

```diff
@@ -0,0 +1,244 @@
+// Libraries
+import React from 'react';
+import clonedeep from 'lodash.clonedeep';
+
+// Partial components
+import bem from 'js/bem';
+import Modal from 'js/components/common/modal';
+import ChartTypePicker from './chartTypePicker.component';
+import ChartColorsPicker from './chartColorsPicker.component';
+import Button from 'js/components/common/button';
+import ReportsModalTabs, {DEFAULT_REPORTS_MODAL_TAB} from 'js/components/reports/reportsModalTabs.component';
+
+// Utilities
+import {actions} from 'js/actions';
+import {handleApiFail} from 'js/api';
+
+// Types & constants
+import {
+  type ReportStyleName,
+  type ReportStyle,
+  DEFAULT_MINIMAL_REPORT_STYLE,
+} from './reportsConstants';
+import type {ReportsState} from './reports';
+import type {ReportsModalTabName} from 'js/components/reports/reportsModalTabs.component';
+import type {FailResponse} from 'js/dataInterface';
+
+interface QuestionGraphSettingsProps {
+  parentState: ReportsState;
```

Comment:
I feel like it would be too much work at this point. It seems like it would be easier to rewrite a lot of things. I will add a comment and wait for some future feature in reports to do bigger refactor